wweval/R/supplement_analyses_and_figs.R

```diff
@@ -0,0 +1,728 @@
+
+    ggsave(fig,
+      filename = file.path(fig_file_dir, "forecast_and_score_comp_fig.png"),
+      width = 7, height = 10
+    )
+
+    ggsave(fig,
+      filename = file.path(
+        alt_fig_file_dir,
+        glue::glue("comp_{this_forecast_date}.png")
+      )
+    )
+  } else {
+    fig <- NULL
+  }
+
+  return(fig)
+
+
+
+  # Also make and save a figure with the full calibration period
+  p_forecasts_all <- ggplot(quantiles_wide) +
+    geom_point(aes(x = date, y = eval_data),
+      fill = "white", size = 1, shape = 21,
+      show.legend = FALSE
+    ) +
+    geom_point(
+      aes(x = date, y = calib_data),
+      color = "black", show.legend = FALSE
+    ) +
+    geom_line(
+      aes(
+        x = date, y = `0.5`,
+        color = model_type
+      )
+    ) +
+    geom_ribbon(
+      aes(
+        x = date, ymin = `0.025`, ymax = `0.975`,
+        fill = model_type
+      ),
+      alpha = 0.1
+    ) +
+    geom_ribbon(
+      aes(
+        x = date, ymin = `0.25`, ymax = `0.75`,
+        fill = model_type
+      ),
+      alpha = 0.2,
+    ) +
+    geom_vline(aes(xintercept = lubridate::ymd(this_forecast_date)),
+      linetype = "dashed"
+    ) +
+    scale_x_date(
+      date_breaks = "1 week",
+      labels = scales::date_format("%Y-%m-%d")
+    ) +
+    xlab("") +
+    ylab("Daily hospital admissions") +
+    scale_color_manual(values = colors$model_colors) +
+    scale_fill_manual(values = colors$model_colors) +
+    get_plot_theme(x_axis_dates = TRUE) +
+    theme(
+      legend.position = "top",
+      legend.justification = "left"
+    ) +
+    labs(color = "Model", fill = "Model") +
+    ggtitle(glue::glue("{this_forecast_date} in {this_location}"))
+
+  ggsave(p_forecasts_all,
+    filename = file.path(fig_file_dir, "calib_and_forecasts.png"),
+    width = 7, height = 10
+  )
+
+  ggsave(p_forecasts_all,
+    filename = file.path(
+      alt_fig_file_dir,
+      glue::glue("calib_and_forecasts_{this_forecast_date}.png")
+    )
+  )
+}
+
+get_plot_wis_t <- function(hosp_quantiles,
+                           scores,
+                           eval_output_subdir,
+                           submissions_path = "https://raw.githubusercontent.com/reichlab/covid19-forecast-hub/master/data-processed/", # nolint
+                           truth_data_path = "https://media.githubusercontent.com/media/reichlab/covid19-forecast-hub/master/data-truth/truth-Incident%20Hospitalizations.csv", # nolint
+                           hub_comparison_model = "COVIDhub-4_week_ensemble") {
+  truth_data <- truth_data <- readr::read_csv(truth_data_path)
+
+  this_location <- hosp_quantiles |>
+    dplyr::distinct(location) |>
+    dplyr::pull()
+
+  loc_code <- cfaforecastrenewalww::loc_abbr_to_flusight_code(this_location)
+
+  this_forecast_date <- hosp_quantiles |>
+    dplyr::distinct(forecast_date) |>
+    dplyr::pull()
+  scores <- scores |> dplyr::filter(
+    forecast_date == !!this_forecast_date,
+    location == !!loc_code
```

Comment:
I don't think the `!!` is necessary.

Reply:
Disagree. Additional column names of `scores` aren't known with certainty, so good to evaluate the variables explicitly just in case.